server/src/test/java/io/crate/expression/reference/doc/lucene/SourceParserTest.java

```diff
@@ -314,42 +313,34 @@ public void test_nested_arrays_from_ignored_objects() {
 
     @Test
     public void test_dropped_leaf_sub_column() {
+        String deleted = "ob";
         SourceParser sourceParser = new SourceParser(
-            Set.of(
-                createReference(ColumnIdent.of("o", List.of("oo", "b")), DataTypes.INTEGER),
-                createReference(ColumnIdent.of("o", List.of("oo", "s")), DataTypes.INTEGER)
-            ),
-            UnaryOperator.identity(),
+            oidOrName -> {
+                return oidOrName.equals(deleted) ? null : oidOrName;
+            },
             true
         );
         var ooType = ObjectType.of(ColumnPolicy.DYNAMIC)
-            .setInnerType("a", DataTypes.INTEGER)
-            .setInnerType("b", DataTypes.INTEGER)
-            .setInnerType("s", DataTypes.INTEGER)
-            .setInnerType("t", DataTypes.INTEGER)
+            .setInnerType("oa", DataTypes.INTEGER)
```

Comment:
I changed the name of some of these because for tables >= 5.5 the key in the stored source would be the oid and must be unique - otherwise the `lookupName` function can't disambiguate top/sub-columns in regards if it is deleted or not.